services/speech_to_text/v1.js

```diff
@@ -74,121 +216,105 @@ module.exports = function (RED) {
+        temp.open({suffix: '.' + fileType(msg.payload).ext}, function (err, info) {
+          if (err) {
+            this.status({fill:'red', shape:'ring', text:'unable to open audio stream'});          
+            var message ='Node has been unable to open the audio stream'; 
 
-        }
+            node.error(message, msg);
+            return;        
+          }  
 
+          stream_buffer(info.path, msg.payload, function (format) {
+            var audio = fs.createReadStream(info.path);
 
-        node.status({fill:'blue', shape:'dot', text:'requesting'});
-        speech_to_text.recognize(params, function (err, res) {
-          node.status({})
-          if (err) {
-            node.error(err, msg);
-          } else {
-            msg.transcription = '';
-            if (res.results.length && res.results[0].alternatives.length) {
-              msg.transcription = res.results[0].alternatives[0].transcript;
-            }
-          }
-
-          node.send(msg);
-          if (cb) {
-            cb();
-          }
+            performAction(audio, format, actionComplete, temp.cleanup);        
+          });
         });
-      };
-
-      var stream_buffer = function (file, contents, cb) {
-        fs.writeFile(file, contents, function (err) {
+      } else if (urlCheck(msg.payload)) {
+        temp.open({suffix: '.audio'}, function(err, info){
           if (err) {
-            throw err;
-          }
-          cb(fileType(contents).ext)
-        });
-      };
-
-      var stream_url = function (file, location, cb) {
-        var wstream = fs.createWriteStream(file);
-
-        wstream.on('finish', function () {
-          fs.readFile(file, function (err, buf) {
-            if (err) {
-              throw err;
-            }
-            cb(fileType(buf).ext)
-          })
-        });
-
-        request(location)
-        .pipe(wstream);
-      };
+            this.status({fill:'red', shape:'ring', 
+              text:'unable to open url audio stream'});          
+            var message ='Node has been unable to open the url audio stream'; 
 
-      temp.open({suffix: '.audio'}, function (err, info) {
-        if (err) throw err;
+            node.error(message, msg);
+            return;        
+          }  
 
-        var stream_payload = (typeof msg.payload === 'string') ? stream_url : stream_buffer;
+          stream_url(info.path, msg.payload, function (format) {
+            var audio = fs.createReadStream(info.path);
 
-        stream_payload(info.path, msg.payload, function (format) {
-          s2t(fs.createReadStream(info.path), format, temp.cleanup);
+            performAction(audio, format, actionComplete, temp.cleanup);        
+          });
         });
-      });
-    });
+      } else {
+        this.status({fill:'red', shape:'ring', text:'payload is invalid'});          
+        var message = 'Payload must be either an audio buffer or a string representing a url'; 
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Variable 'message' is already declared in line 246.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2636145243/issues/source?bid=3225115&fileBranchId=3319654#l305)